CHANGES.md

```diff
@@ -1,5 +1,22 @@
 # Change log for kotlinx.coroutines
 
+## Version 1.2.1
+
+Major:
+  * Infrastructure for testing coroutine-specific code in `kotlinx-coroutines-test`: `runBlockingTest`, `TestCoroutineScope` and `TestCoroutineDispatcher`, contributed by Sean McQuillan (@objcode).
+  * `Job.asCompletableFuture` extension in jdk8 module (#1113).
+
+Flow improvements:
+  * `flowViaChannel` rework: block parameter is no longer suspending, but provides `CoroutineScope` receiver and allows conflated channel (#1081, #1112).
+  * New operators: `switchMap`, `sample`, `debounce` (#1107).
+  * `consumerEach` is deprecated for removal on `Publisher`, `ObservableSource` and `MaybeSource`, `collect` extension is introduced instead (#1080).
+
+Other:
+  * Race in Job.join and concurrent cancellation is fixed (#1123).
+  * Stacktrace recovery machinery improved.
+  * Atomicfu-related bug fixes: publish transformed sources, do not propagate transitive atomicfu dependency (#1064, #1116).
```

Comment:
"publish transformed source" -> "publish transformed artifacts" (otherwise it sounds ambiguous)